client/extensions/woocommerce/app/dashboard/setup-tasks.js

```diff
@@ -60,9 +61,12 @@ class SetupTasks extends Component {
 
 		if ( site && site.ID ) {
 			this.props.fetchPaymentMethods( site.ID );
-			this.props.fetchProducts( site.ID, 1 );
 			this.props.fetchSettingsGeneral( site.ID );
 			this.props.fetchSetupChoices( site.ID );
+
+			if ( ! areProductsLoaded ) {
```

Comment:
The reason for this, is fetchProducts always re-fetches (a decision we made for pagination), and the two dashboard pages can get stuck in a loop if we try to load multiple times.